swig/include/Dataset.i

```diff
@@ -889,6 +889,38 @@ CPLErr AdviseRead(  int xoff, int yoff, int xsize, int ysize,
 
 #endif /* defined(SWIGPYTHON) || defined(SWIGJAVA) */
 
+#ifdef SWIGCSHARP
+  
+  %newobject GetNextFeature;
+  OGRFeatureShadow *GetNextFeature( OGRLayerShadow** ppoBelongingLayer = NULL,
+                                    double* pdfProgressPct = NULL,
+                                    GDALProgressFunc callback = NULL,
+                                    void* callback_data=NULL )
+  {
+    return GDALDatasetGetNextFeature( self, ppoBelongingLayer, pdfProgressPct,
+                                      callback, callback_data );
+  }
+
+  int GetLayerCount() {
+    return GDALDatasetGetLayerCount(self);
+  }
+
+  OGRLayerShadow *GetLayer(int index ) {
+    OGRLayerShadow* layer = (OGRLayerShadow*) GDALDatasetGetLayer(self, index);
+    return layer;
+  }
+
+  OGRLayerShadow *GetLayerByName(const char* layerName) {
+    OGRLayerShadow* layer = (OGRLayerShadow*) GDALDatasetGetLayerByName(self, layerName);
+    return layer;
+  }
+
+  void ResetReading()
+  {
+    GDALDatasetResetReading(self);
+  }
+
```

Comment:
those already exist in ``#if defined(SWIGPYTHON) || defined(SWIGJAVA)`` branch at line 772. It would be better to move the existing definitions out of that branch, to have it in common for all bindings.  That would make the GetNextFeature() definition the only SWIGCSHARP specific one

Reply:
@rouault I agree. Updated the PR. Something like that?

Reply:
yes, looks good. I've adjusted indentation